cmd/scollector/collectors/sql_windows.go

```diff
@@ -85,24 +91,32 @@ func c_mssql() (opentsdb.MultiDataPoint, error) {
 	return md, err
 }
 
-func c_mssql_general() (opentsdb.MultiDataPoint, error) {
-	var dst []Win32_PerfRawData_MSSQLSERVER_SQLServerGeneralStatistics
-	var q = wmi.CreateQuery(&dst, `WHERE Name <> '_Total'`)
-	if err := queryWmi(q, &dst); err != nil {
-		return nil, err
-	}
+func c_mssql_general(svc_dst []Win32_Service) (opentsdb.MultiDataPoint, error) {
 	var md opentsdb.MultiDataPoint
-	for _, v := range dst {
-		Add(&md, "mssql.user_connections", v.UserConnections, nil, metadata.Gauge, metadata.Count, descMSSQLUserConnections)
-		Add(&md, "mssql.connection_resets", v.ConnectionResetPersec, nil, metadata.Counter, metadata.PerSecond, descMSSQLConnectionResetPersec)
-		Add(&md, "mssql.logins", v.LoginsPersec, nil, metadata.Counter, metadata.PerSecond, descMSSQLLoginsPersec)
-		Add(&md, "mssql.logouts", v.LogoutsPersec, nil, metadata.Counter, metadata.PerSecond, descMSSQLLogoutsPersec)
-		Add(&md, "mssql.mars_deadlocks", v.MarsDeadlocks, nil, metadata.Counter, metadata.Count, descMSSQLMarsDeadlocks)
-		Add(&md, "mssql.proc_blocked", v.Processesblocked, nil, metadata.Gauge, metadata.Count, descMSSQLProcessesblocked)
-		Add(&md, "mssql.temptables_created", v.TempTablesCreationRate, nil, metadata.Counter, metadata.PerSecond, descMSSQLTempTablesCreationRate)
-		Add(&md, "mssql.temptables_to_destroy", v.TempTablesForDestruction, nil, metadata.Gauge, metadata.Count, descMSSQLTempTablesForDestruction)
-		Add(&md, "mssql.transactions_total", v.Transactions, nil, metadata.Gauge, metadata.Count, descMSSQLTransactions)
+	for _, w := range svc_dst {
+		var dst []Win32_PerfRawData_MSSQLSERVER_SQLServerGeneralStatistics
+		var q = wmi.CreateQuery(&dst, `WHERE Name <> '_Total'`)
```

Comment:
Would use `q :=` and `label :=`. Generally var isn't used if you are assigning a value.